app/__init__.py

```diff
@@ -78,10 +79,11 @@ def create_app(application, config=None):
 
     application.config["NOTIFY_APP_NAME"] = application.name
     init_app(application)
+    application.json_encoder = NotifyJSONEncoder
```

Comment:
It wasn't immediately clear to me how this all works, but after review I think that this sets a custom json encoder in flask which is then called whenever we attempt to encode JSON using `flask.jsonify()`?

Reply:
Yes that's it. Flask offers its own abstraction that will make use of the regular Python `json` module, but we need to short-circuit it by specifying our own which simply overrides the existing ones, enriching it with the custom types we're interested in.

To add to the confusion, Notify does not always call on `jsonify` but makes use of marshmallow sometimes. It does not have a consistent JSON serialization strategy, depending if the object comes from the model or not.